### PR TITLE
fix: handling errors for non-existing endpoints

### LINKED ISSUE #4073: [BUG-python/deployment] backwards compatibility `src/argilla/client/sdk/v1/datasets/api.py` `get_metadata_properties`
**Describe the bug**
Whenever retrieving a dataset from server 1.17 with client 1.19 there seems to be an incompatibility issue when calling `response.json`. I think this might be a more recurrent issue, where for some reason the response does return with `response.status_code==200` even though the endpoint is not defined in FastAPI. Hence this might also occur with the semantic search update. 

**Stacktrace and Code to create the bug**
https://github.com/argilla-io/argilla/blob/38c279984b007576880118e2ab8b66451bac60b5/src/argilla/client/sdk/v1/datasets/api.py#L423
Try loading a dataset from the 1.17 server with a 1.19 client.

**Expected behavior**

**Environment:**
 - Argilla Version [e.g. 1.0.0]: 1.19
 - ElasticSearch Version [e.g. 7.10.2]: N.A.
 - Docker Image (optional) [e.g. argilla:v1.0.0]: 1.17

**Additional context**
N.A.

### PR DESCRIPTION
**Pull request by frascuchon**
# Description

This PR adds a callback controller to return a proper status code and error message when a client requests a non-existing endpoint.

This change will help identify incompatibility problems when using new clients over old server instances as suggested in #4073 

Closes #4073

**Type of change**

(Please delete options that are not relevant. Remember to title the PR according to the type of change)

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

**How Has This Been Tested**

Calling unexisting endpoints locally 

**Checklist**

- [X] I followed the style guidelines of this project
- [X] I did a self-review of my code
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK) (see text above)
- [ ] I have added relevant notes to the `CHANGELOG.md` file (See https://keepachangelog.com/)

### CODE REVIEW

File: src/argilla/server/routes.py [hunk 1]
```diff
@@ -76,3 +76,8 @@
 api_router.include_router(users_v1.router, prefix="/v1")
 api_router.include_router(vectors_settings_v1.router, prefix="/v1")
 api_router.include_router(workspaces_v1.router, prefix="/v1")
+
+
+@api_router.route("/{full_path:path}")
```
Comment (davidberenstein1957) [inline src/argilla/server/routes.py]: does this handle both GET POST etc?

File: src/argilla/server/routes.py [hunk 2]
```diff
@@ -76,3 +76,8 @@
...
+def endpoint_not_found_controller(request: Request):
+    raise HTTPException(status_code=404, detail=f"Endpoint {request.url.path!r} not found")
```
Comment (davidberenstein1957) [inline src/argilla/server/routes.py:83]: maybe we can urge mention potential server and client version misalignment?

Comment (jfcalvo) [inline src/argilla/server/routes.py:83]: Maybe it's interesting to don't add this to swagger API schema.

**Review by jfcalvo — APPROVED** [review]

File: src/argilla/server/routes.py (hunk 1 above)
Comment (frascuchon) [reply src/argilla/server/routes.py]: Yes. Done. Thanks!

File: src/argilla/server/routes.py (hunk 2 above)
Comment (frascuchon) [reply src/argilla/server/routes.py:83]: Done. Using `include_in_schema=False`

Comment (frascuchon) [reply src/argilla/server/routes.py:83]: This error should be processed on the client side. The server cannot assume that these kinds of errors are related to a misalignment between the server and the client.

**Review by davidberenstein1957 — APPROVED** [review]

**Pull request merged**